specification/monitor/resource-manager/Microsoft.Insights/preview/2018-11-27-preview/vmInsightsOnboarding_API.json

```diff
@@ -0,0 +1,260 @@
+            "type": "string",
+            "description": "Error code identifying the specific error."
+          },
+          "message": {
+            "type": "string",
+            "description": "Error message in the caller's locale."
+          }
+        },
+        "required": [
+          "code"
+        ]
+      },
+      "WorkspaceInfo": {
+        "type": "object",
+        "description": "Information about a Log Analytics Workspace.",
+        "properties": {
+          "id": {
+            "type": "string",
+            "description": "Azure Resource Manager identifier of the Log Analytics Workspace."
+          },
+          "location": {
+            "type": "string",
+            "description": "Location of the Log Analytics workspace."
+          },
+          "properties": {
+            "x-ms-client-flatten": true,
+            "type": "object",
+            "description": "Resource properties.",
+            "properties": {
+              "customerId": {
+                "type": "string",
+                "description": "Log Analytics workspace identifier."
+              }
+            },
+            "required": [
+              "customerId"
+            ]
+          }
+        },
+        "required": [
+          "id",
+          "location",
+          "properties"
+        ]
+      },
+      "DataContainer": {
+        "type": "object",
+        "description": "Information about a container with data for a given resource.",
+        "properties": {
+          "workspace": {
+            "$ref": "#/definitions/WorkspaceInfo",
+            "description": "Log Analytics workspace information."
+          }
+        },
+        "required": [
+          "workspace"
+        ]
+      },
+      "VMInsightsOnboardingStatus": {
+        "description": "VM Insights onboarding status for a resource.",
+        "allOf": [
+          {
+            "$ref": "#/definitions/ProxyResource"
+          }
+        ],
+        "properties": {
+          "properties": {
+            "x-ms-client-flatten": true,
+            "type": "object",
+            "description": "Resource properties.",
+            "properties": {
+              "resourceId": {
+                "type": "string",
+                "description": "Azure Resource Manager identifier of the resource whose onboarding status is being represented."
+              },
+              "onboardingStatus": {
+                "type": "string",
+                "enum": [
+                  "onboarded",
+                  "notOnboarded",
+                  "unknown"
+                ],
+                "x-ms-enum": {
+                  "name": "OnboardingStatus",
+                  "modelAsString": true
+                },
+                "description": "The onboarding status for the resource. Note that, a higher level scope, e.g., resource group or subscription, is considered onboarded if at least one resource under it is onboarded."
+              },
+              "dataStatus": {
```

Comment:
So, dataStatus is an aggregate in the case of multiple destinations? I.e. if the VM were onboarded to multiple destinations, and only sending to one, it would still be present? Is this enough fidelity or should the status be a property of the destination?